### PR TITLE
"raise StopIteration" replaced to "return" 

### PR DESCRIPTION
**Pull request by SilentFlame**
all the "raise StopIteration" are replaced with "return".
#10134

some of the ambiguous replacements are still left like:
`raise SqrtdenestStopIteration` which is a subclass here.

### CODE REVIEW

File: sympy/strategies/branch/core.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def chain(*brules):
     def chain_brl(expr):
         if not brules:
             yield expr
-            raise StopIteration()
+            return()
```
Comment (asmeurer) [inline sympy/strategies/branch/core.py]: This is wrong. It should just be `return`.

**Comment by VinaySingh035** [comment]
ok going to change it.

**Comment by asmeurer** [comment]
Looks like the code in factor_ (and maybe other places) will have to be refactored, since it catches the StopIteration that is being raised.

**Comment by jksuom** [comment]
I am not sure that it would be necessary to change the use of StopIteration in factor_ as it is not connected with generators. Refactoring seems like a bigger operation.

**Comment by asmeurer** [comment]
Does the use in factor_ result in a RuntimeError with the new `from __future__ import generator_stop`? If so, it should be fixed. If not, then it's fine.

**Comment by kshitij10496** [comment]
@VinaySingh035 Are you willing to work on this ?
If you are busy, I can give this a try

**Comment by czgdp1807** [comment]
The issue has already been fixed some PRs like, #10591 and #11649
Therefore closing this PR. Please let us know if there is a need to re-open this. Thanks for the contributions.

**Pull request closed without merging**